include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -1300,72 +1339,63 @@ void solveRank1ProblemDist(CommSender&& row_comm, CommSender&& col_comm, const S
           // Note:
           // Considering that
           // - LAED4 requires working on k elements
-          // - Weight computation requires working on m_subm_el_lc
+          // - Weight computation requires working on m_el_lc
           //
           // and they are needed at two steps that cannot happen in parallel, we opted for allocating
           // the workspace with the highest requirement of memory, and reuse them for both steps.
-          const SizeType max_size = std::max(k, m_subm_el_lc);
+          const SizeType max_size = std::max(k, m_el_lc);
           for (std::size_t i = 0; i < nthreads; ++i)
             ws_cols.emplace_back(max_size);
-          ws_cols.emplace_back(m_subm_el_lc);
+          ws_cols.emplace_back(m_el_lc);
 
-          ws_row = memory::MemoryView<T, Device::CPU>(n_subm_el_lc);
-          std::fill_n(ws_row(), n_subm_el_lc, 0);
+          ws_row = memory::MemoryView<T, Device::CPU>(n_el_lc);
+          std::fill_n(ws_row(), n_el_lc, 0);
         }
 
         // Note: we have to wait that LAED4 workspaces are ready to be used
         barrier_ptr->arrive_and_wait(barrier_busy_wait);
 
-        const T* d_ptr = d_tiles_futs[0].get().ptr();
+        const T* d_ptr = d_tiles[0].get().ptr();
         const T* z_ptr = z_tiles[0].ptr();
 
         // STEP 1: LAED4 (multi-thread)
-        {
-          common::internal::SingleThreadedBlasScope single;
+        if (thread_idx == 0) {  // TODO make it multi-threaded over multiple workers
+          common::internal::SingleThreadedBlasScope single;  // TODO needed also for laed?
```

Comment:
@msimberg is this needed also for lapack calls, right?

Reply:
Absolutely. I was torn about the name when I added it, so if you have suggestions for a name that would describe it better we can consider renaming it (not here though...).